### ISSUE TITLE
displaymanager module bug "'module' object has no attribute 'scandir'"

### ISSUE THREAD

**Issue by geckolinux** [calamares/calamares#804]
Hi, this bug seems to have just appeared with 3.1.4. This is for an openSUSE 42.3 spin using the displaymanager module and the `sysconfigSetup: true` option with lightDM:

```Boost.Python error in job "displaymanager".
<class 'AttributeError'>
'module' object has no attribute 'scandir'
Traceback:
  File "/usr/lib64/calamares/modules/displaymanager/main.py", line 486, in run
    for entry in os.scandir(greeter_path):
```
I also tried putting `lightdm` first in the list of `displaymanagers`, but it made no difference. This results in a failed installation.

I'm also attaching the /root/.cache/Calamares/calamares.log [here](https://github.com/calamares/calamares/files/1307690/calamares.log.txt) .

**Comment by philmmanjaro** [comment]
The code was submitted with https://github.com/calamares/calamares/commit/a1173bccc2fa7b512d573d77cb0e3b0e61615949 and was therefore already present in **v3.1.2**. Seems Suse 42.3 has a different **python3** version. There seems to be some issue with function `os.scandir`. See also [here](https://docs.python.org/3/library/os.html#os.scandir).

**Comment by geckolinux** [comment]
Hi @philmmanjaro  and thanks for the fast reply. I'm not sure why this changed, as it was working with the latest Calamares master build that I tested on openSUSE 42.3 just last week.

**Comment by philmmanjaro** [comment]
Please check the package list, which might had changed. Also you can write a small **python3** script to test the given function.

**Comment by geckolinux** [comment]
The python3 version for openSUSE 42.3 is at 3.4.6-11.1, and it hasn't received any updates since 42.3 was release months ago.

**Comment by philmmanjaro** [comment]
Arch and Manjaro are at `python 3.6.2-1`. [scandir](https://pypi.python.org/pypi/scandir) was included in **v3.5** into **python**. Maybe you package it for Suse as a separate package. Seems most [struggle](https://www.reddit.com/r/openSUSE/comments/6a86ql/updating_python_3_on_leap_422/) in general with it. Another option would be to patch the code to a similar function which is available for **v3.4** series, e.p. [os.walk](https://docs.python.org/3.4/library/os.html#os.walk).

**Comment by adriaandegroot** [comment]
That code just needs os.listdir(), which has been fixed in c8fd11b103b10d2b1a78ee7bde3dd8cc7ebee869

**Comment by geckolinux** [comment]
@adriaandegroot Thanks a lot for the fast fix, sorry I couldn't respond sooner.

I tested Calamares master from 2017-09-20, and now I am getting a different error:

```
Boost.Python error in job "displaymanager".
<class 'AttributeError'>
'str' object has no attribute 'name'
Traceback:
  File "/usr/lib64/calamares/modules/displaymanager/main.py", line 486, in run
    if entry.name.endswith('.desktop'):
```
Sorry that I appear to have lost the installation log from `/root/.cache`, but I can reproduce the error again if you need that too.

Thanks a lot for the help.

**Comment by philmmanjaro** [comment]
It creates a regression with `'str' object has no attribute 'name'`. So we have to fix it before we release a next stable release. On my end I've reverted it so far as **python 3.6** is just fine with the original code.

**Comment by kkofler** [comment]
I think `entry.name` probably needs to be changed to just `entry`.

**Comment by philmmanjaro** [comment]
Fixed with d20285d4f20a3f7ff76ea145b50bdec0b7d0ef2f
